client/src/routes/challs.js

```diff
@@ -0,0 +1,126 @@
+import { useCallback, useState, useEffect } from 'preact/hooks'
+
+import config from '../../../config/client'
+import withStyles from '../components/jss'
+import Problem from '../components/problem'
+
+import { getChallenges, getPrivateSolves } from '../api/challenges'
+
+const Challenges = ({ classes }) => {
+  const [problems, setProblems] = useState([])
+  const [categories, setCategories] = useState({})
+  const [showSolved, setShowSolved] = useState(false)
+  const [solveIDs, setSolveIDs] = useState([])
+
+  const setSolvedFn = useCallback(id => () => {
+    if (!solveIDs.includes(id)) {
+      setSolveIDs([...solveIDs, id])
+    }
+  }, [solveIDs])
+
+  const handleInvertShowSolved = useCallback(() => {
+    setShowSolved(!showSolved)
+  }, [showSolved])
+
+  const handleInvertCategoryStateFn = prop => () => {
```

Comment:
This is creating a new handler function every render, consider refactoring the checkboxes into components that take the category as a prop and then pass the prop into an updater function (the function returned by this closure would work, just remove the `() =>` and wrap in `useCallback`)